## Assignment 3

This assignmemt is based on content discussed in module 3 and test basic concepts of probability theory and normalization in statistics.

## Learning outcomes

-   Work on problems of different distributions eg., binomial, gaussian 
-   Calculate z score 
-	 Make statistical inferences on given data
-	 Construct a null and an alternate hypothesis
-	 Find the p-value for a given hypothesis and T test statistic.


**Question 1**

The Capital Asset Pricing Model (CAPM) is a financial model that assumes returns on a portfolio are normally distributed.  Suppose a portfolio has an average annual return of 14.7% (i.e., an average gain on 14.7%) with a standard deviation of 33%.  A return of 0% means the value of the portfolio doesn't change, a negative return means that the portfolio loses money, and a positive return means that the portfolio gains money. Determine the following:

1. What percentage of years does this portfolio lose money, (i.e. have a return less than 0%)?
2. What is the cutoff for the highest 15% of annual returns with this portfolio?

See CAPM here https://en.wikipedia.org/wiki/Capital_asset_pricing_model 

In [1]:
import scipy.stats
import math

In [2]:
#Annual Return of 14.7%
#Standard Deviation of 33%

#1) What percentage of years does this portfolio lose money (i.e. have a return less than 0%

scipy.stats.norm(14.7,33).cdf(0)

0.3279956507031998

In about 32.8% of the years the portfolio will have a negative return

In [3]:
#2) What is the cutoff for the highest 15% of annual returns with this portfolio

scipy.stats.norm(14.7,33).ppf(0.85)

48.90230185329506

15% of the returns will be 48.9% or higher

**Question 2**

Past experience indicates that because of low morale, a company loses 20 hours a year per employee due to lateness and abstenteeism.  Assume that the standard deviation of the population is 6 and normally distributed.

The HR department implemented a new rewards system to increase employee morale, and after a few months it collected a random sample of 20 employees and the annualized absenteeism was 14.

1. Could you confirm that the new rewards system was effective with a 90% confidence?
2. An HR subject matter expert would be very happy if the program could reduce absenteeism by 20% (i.e. to 16 hours).  Given the current sampling parameters, what is the probability that the new rewards system reduced absenteeism to 16 hours and you miss it?
3. Repeat part 1) and 2) with an α = 95% CI.
4. Based on the answers in 3), is the sampling method good enough to identify a reduction from 20 to 16 hours if I use a confidence of 95%?
5. What should the sample size be if you want β to be 5%

In [4]:
#Company loses 20 hours a year per employee 
#std of the population is 6
#Normally distributed 

#Random sample of 20 employees
#Annualized absenteeism was 14 

#1) New rewards system was effective with a 90% confidence

(scipy.stats.t.cdf(((14-20)/(6/math.sqrt(20))),df=19))*100

0.013059668917013078

0.01% is a very small number, so you can not confirm that the new rewards system was effective with 90% confidence

In [25]:
#2) Reduced absenteeism to 16 hours

1 - scipy.stats.t.pdf(((16-20)/(6/math.sqrt(20))),df=19)

0.9915194858448683

The p value > confidence thus we accept the alternative hypothesis which is that the new system is not effective.
Meaning we probably didn't miss that it reduced absenteeism to 16

In [26]:
#3 / #4 
print((scipy.stats.t.cdf(((14-20)/(6/math.sqrt(20))),df=19))*100)
print(1-(scipy.stats.t.pdf(((16-20)/(6/math.sqrt(20))),df=19)))

0.013059668917013078
0.9915194858448683


The results from above remain the same with a confidence interval of 95%. Note the sampling methods from above are not good enough based on the probability result being less than 5% for both 

In [50]:
#5) Sample size for beta to be 5%
prob = scipy.stats.t.cdf(((14-20)/(6/math.sqrt(20))),df=19)
z_score = scipy.stats.t.ppf(prob, df=19)
z_score_beta = scipy.stats.t.cdf(1-0.05, df=19)
(((z_score_beta + z_score)**2)*(6)**2)/(20)**2

1.1984708372323225

If we want it to be 5%, the sample size should be around 1

**Question 3**

Chi-Square Goodness of fit

Please access and review **section 6.3.5** in the OpenIntro Statistics textbook:

Diez, D., Barr, C. & Çetinkaya-Rundel, M. (2017). OpenIntro Statistics (3rd Ed.). https://www.openintro.org/stat/textbook.php?stat_book=os

Given the information in section 6.3.5, write python code for the following:

 - Calculate the expected values based on the geometric distribution with a probability of 53.2%
 - Compare the expected vs. the observed values from the textbook using the Chi-Square distribution
 - Reach a conclusion
 - Explain what is the business impact of your conclusion

In [69]:
import pandas as pd
#your work here
#Days -> Observed
#1 -> 717, 2 -> 369, 3 -> 155, 4 -> 69, 5 -> 28, 6 -> 14, 7+ -> 10, total = 1362
observed = {'Days':[1,2,3,4,5,6,7], 
               'Observed':[717,369,155,69,28,14,10]}
df_observed = pd.DataFrame(observed)
#Days -> Geometric
#1 -> 743, 2 -> 338, 3 -> 154, 4 -> 70, 5 -> 32, 6 -> 14, 7+ -> 12, total = 1362

In [65]:
#Calculating expected values based on geometric distribution w a prod of 53.2%
print("Expected value for Day 1:",(((1-0.532)**(1-1))*(0.532))*1362)
print("Expected value for Day 2:",(((1-0.532)**(2-1))*(0.532))*1362)
print("Expected value for Day 3:",(((1-0.532)**(3-1))*(0.532))*1362)
print("Expected value for Day 4:",(((1-0.532)**(4-1))*(0.532))*1362)
print("Expected value for Day 5:",(((1-0.532)**(5-1))*(0.532))*1362)
print("Expected value for Day 6:",(((1-0.532)**(6-1))*(0.532))*1362)
print("Expected value for Day 7+:",(((1-0.532)**(7-1))*(0.532))*1362)

Expected value for Day 1: 724.5840000000001
Expected value for Day 2: 339.105312
Expected value for Day 3: 158.70128601599998
Expected value for Day 4: 74.27220185548799
Expected value for Day 5: 34.75939046836838
Expected value for Day 6: 16.2673947391964
Expected value for Day 7+: 7.613140737943915


In [81]:
#Compare the expected vs. the observed values using the Chi-Square Distribution 
expected = {'Days':[1,2,3,4,5,6,7], 
               'Observed':[725,339,159,74,35,16,8]}
df_expected = pd.DataFrame(expected)

chi2_stat, p_val, dof, ex = scipy.stats.chi2_contingency(df_expected)
print("For expected: chi2_stat of",chi2_stat," p_value of ", p_val, "degrees of freedom of ", dof)
chi2, p, df, ex = scipy.stats.chi2_contingency(df_observed)
print("For observed: chi2_stat of",chi2," p_value of ", p, "degrees of freedom of ", df)

For expected: chi2_stat of 264.1192095146861  p_value of  3.9294282492745334e-54 degrees of freedom of  6
For observed: chi2_stat of 261.63674099269497  p_value of  1.3342906314572928e-53 degrees of freedom of  6


The null hypothesis is that the stock market being up or down on a given day is independent from all other days.
The p value of both the expected and observed is less than 0.05 there we reject the null hypothesis

What does rejecting the null hypothesis mean from a business impact perspective?
This means that the stock market being up or down on a given day depends on the other days as market conditions can continue day over day for longer periods. 